src/Microsoft.Graph.Core/Requests/Content/BatchRequestContent.cs

```diff
@@ -29,6 +30,8 @@ public class BatchRequestContent: HttpContent
         /// </summary>
         public IReadOnlyDictionary<string, BatchRequestStep> BatchRequestSteps { get; private set; }
 
+        private ArrayList OrderedBatchRequestIds { get; set; }
```

Comment:
I'm abit unsure about having to double track two structures at a time when adding/remove items as it could be a gotcha if the code evolves. 

Maybe we could have an internal implementation of `IDictionary` similar to https://github.com/microsoft/kiota-dotnet/blob/main/src/abstractions/RequestHeaders.cs and override its enumerator

This way, when we get to .NET 9 we can simply swap the class name.

Thoughts?